extensions/lifecycle/tasks/LifecycleColdStatusArchiveTask.js

```diff
@@ -0,0 +1,147 @@
+const async = require('async');
+const { errors } = require('arsenal');
+
+const ObjectMDArchive = require('arsenal').models.ObjectMDArchive;
+const LifecycleUpdateTransitionTask = require('./LifecycleUpdateTransitionTask');
+
+class LifecycleColdStatusArchiveTask extends LifecycleUpdateTransitionTask {
+    /**
+     * Process a lifecycle object entry
+     *
+     * @constructor
+     * @param {LifecycleObjectProcessor} proc - object processor instance
+     */
+    constructor(proc) {
+        const procState = proc.getStateVars();
+        super();
+        Object.assign(this, procState);
+    }
+
+    getTargetAttribute(entry) {
+        const {
+            bucketName,
+            objectKey,
+            objectVersion,
+            accountId
+        } = entry.getTarget();
+        return {
+            bucket: bucketName,
+            key: objectKey,
+            version: objectVersion,
+            accountId
+        };
+    }
+
+    getSourceAttribute(entry) {
+        const { bucketName, objectKey, objectVersion } = entry.getTarget();
```

Comment:
could spread bucket name and object version to bucket and version variable names directly here